src/interactable_spikes.c

```diff
@@ -0,0 +1,992 @@
+        if (gravityInverted) {
+            if (r4 & 0x10000) {
+                gPlayer.y += (r4 << 24) >> 16;
+                gPlayer.speedAirY = 0;
+                gPlayer.moveState &= (~MOVESTATE_8);
+                gPlayer.moveState |= MOVESTATE_IN_AIR;
+                gPlayer.unk3C = 0;
+            }
+            // _0805FC6C
+
+            if (r4 & 0x20000) {
+                if (sub_8060D08(s, screenX, screenY, &gPlayer) & 0x10000) {
+                    gPlayer.y += (r4 << 24) >> 16;
+                    gPlayer.speedAirY = 0;
+                    gPlayer.moveState |= MOVESTATE_8;
+                    gPlayer.moveState &= ~MOVESTATE_IN_AIR;
+                    gPlayer.unk3C = s;
+                }
+            }
+        } else {
+            // _0805FCC8
+            if (r4 & 0x10000) {
+                if (sub_8060D08(s, screenX, screenY, &gPlayer) & 0x10000) {
+                    gPlayer.y += (r4 << 24) >> 16;
+                    gPlayer.speedAirY = gravityInverted; // Q_8_8(0.5) instead?
+                }
+            }
+
+            if (r4 & 0x20000) {
+                gPlayer.y += (r4 << 24) >> 16;
+                gPlayer.speedAirY = Q_8_8(0);
+            }
+        }
+
+        // _0805FD12
+        if (r4 & 0x40000) {
+#ifdef NON_MATCHING
+            u16 iaIndex;
+#else
+            register u16 iaIndex asm("r0");
+#endif
+            gPlayer.moveState |= MOVESTATE_20;
+            gPlayer.x += (s16)((u32)r4 & 0xFF00);
+            gPlayer.speedAirX = 0;
+            gPlayer.speedGroundX = 0;
+
+            iaIndex = IA__SPIKES__NORMAL_LEFT;
+            if (iaIndex != ia->index) {
+                if (sub_800CBA4(&gPlayer)) {
+                    m4aSongNumStart(SE_SPIKES);
+                }
+            }
+        } else {
+            // _0805FD60
+            if (r4 & 0x80000) {
+#ifdef NON_MATCHING
+                u16 iaIndex;
+#else
+                register u16 iaIndex asm("r0");
+#endif
+                gPlayer.moveState |= MOVESTATE_20;
+                gPlayer.x += (s16)((u32)r4 & 0xFF00);
+                gPlayer.speedAirX = 0;
+                gPlayer.speedGroundX = 0;
+
+                iaIndex = IA__SPIKES__NORMAL_LEFT;
+                if (iaIndex == ia->index) {
+                    if (sub_800CBA4(&gPlayer)) {
+                        m4aSongNumStart(SE_SPIKES);
+                    }
+                }
+            }
+        }
+    }
+    // _0805FDA4
+
+    if ((gGameMode == GAME_MODE_MULTI_PLAYER_COLLECT_RINGS) && (ia->d.sData[0] == 0)
+        && (gUnknown_030053E0 == 0)) {
+        if (spikes->unk3C[0] & 0x20) {
+            gPlayer.moveState &= ~MOVESTATE_20;
+        }
+        // _0805FDD8
+        if (spikes->unk3C[0] & 0x8) {
+            gPlayer.moveState &= ~MOVESTATE_8;
+        }
+    }
+    // _0805FDF0
+    if (IS_OUT_OF_RANGE_OLD(u16, s->x, s->y, (CAM_REGION_WIDTH))) {
+        ia->x = spikes->base.spriteX;
```

Comment:
Have we considered a macro for this yet `SET_SPRITE_DESTROYED(ia, spriteX)`

Reply:
Uh, I mean "RESET_ENTITY_POS" would make more sense? (Yes, the init macro needs to be renamed as well)